models/challenges.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * This file contains wrapper functions to interact with the DB.
+ * This will contain the DB schema if we start consuming an ORM for managing the DB operations
+ */
+
+const Firestore = require('@google-cloud/firestore')
+const firestore = require('../utils/firestore')
+
+const challengesModel = firestore.collection('challenges')
+const userModel = firestore.collection('users')
+
+const cannotSubscribe = 'User cannot be subscribed to challenge'
+const userDoesNotExistError = 'User does not exist. Please register to participate'
+const errorMessage = 'Error getting challenges'
+
+/**
+ * Fetch the challenges
+ * @return {Promise<challengesModel|Array>}
+ */
+
+const fetchChallenges = async () => {
+  try {
+    const challengesSnapshot = await challengesModel.get()
+    return (function () {
+      const challenges = []
+      challengesSnapshot.forEach((challengeDoc) => {
+        challenges.push({
+          id: challengeDoc.id,
+          ...challengeDoc.data()
+        })
+      })
+      return challenges
+    })()
```

Comment:
Do we really need to use IIFE here?

Reply:
Even I was confused when I saw it 😅 
I guess I should just remove line number 24 and 31, right?